tests/test_all_fields.py

```diff
@@ -5,13 +5,103 @@
 from tap_tester.base_suite_tests.all_fields_test import AllFieldsTest
 from base import CloseioBase
 
-class CloseioAllFieldsTest(AllFieldsTest,CloseioBase):
+
+class CloseioAllFieldsTest(AllFieldsTest, CloseioBase):
     """Test that with no fields selected for a stream automatic fields are still replicated"""
 
+    selected_fields = AllFieldsTest.selected_fields
```

Comment:
Should this be `CloseioAllFieldsTests.selected_fields`? It probably doesn't actually makes a difference, but it reads like we're taking `selected_fields` that are predefined in `AllFieldsTest`. Since `AllFieldsTest` is an abstract class, it feels a little odd to use the `selected_fields` property from it instead of the implementation of that class (`CloseioAllFieldsTests`).

Reply:
The selected_fields is defined during setup.  The only reason this line is here is pylint.  This line doesn't actually perform any function except to let the rest of the class know where this is defined.

I could just remove this line.  Is that what you are suggesting?

The abstract classes do a combination of implementing methods that it can and defining which ones need to be defined.